apis/v1alpha2/shared_types.go

```diff
@@ -298,6 +302,14 @@ type Group string
 // +kubebuilder:validation:Pattern=`^[a-zA-Z]([-a-zA-Z0-9]*[a-zA-Z0-9])?$`
 type Kind string
 
+// ObjectName refers to the name of a Kubernetes object.
+// Object names can have a variety of forms, including RFC1123 subdomains,
+// RFC 1123 labels, or RFC 1035 labels.
+//
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:MaxLength=253
+type ObjectName string
```

Comment:
I [followed up with sig-api-machinery](https://groups.google.com/g/kubernetes-sig-api-machinery/c/V-qnyODuMt4) and it sounds like there is a [minimum set of validation we can provide here](https://github.com/kubernetes/kubernetes/blob/b290671b1b28ab4af8164e2b69b334fab584558e/staging/src/k8s.io/apimachinery/pkg/api/validation/path/name.go#L30-L46). It may be a bit difficult to translate into regex validation though, and others recommended against adding validation here. Although that minimum set of validation does not include a max length, 253 seems like a reasonable value to me based on the most common validation + ensure we don't end up with an unbounded list.